### ISSUE TITLE
`hab svc load` with --bind fails mysteriously

### ISSUE THREAD

**Issue by baumanj** [habitat-sh/habitat#5178]
In trying to fix up the builder dev environment, ran into this strange behavior (reproduced by @eeyun):

When [`start-builder`](https://github.com/habitat-sh/builder/blob/master/.studiorc#L133-L137) runs, the first couple services come up ok:

```
[27][default:/src:0]# hab sup status
package                                        type        state  uptime (s)  pid     group
habitat/builder-datastore/7043/20180208190943  standalone  up     2494        128869  builder-datastore.default
core/sccache/0.2.4/20180310001750  standalone  up  546  129429  sccache.default
habitat/builder-router/7114/20180302152133  standalone  up  133  129665  builder-router.default
[28][default:/src:0]# 
```

But `start-builder api` fails:
```
[30][default:/src:1]# RUST_LOG=debug hab svc load habitat/builder-api -s at-once --bind router:builder-router.default
+ RUST_LOG=debug
+ hab svc load habitat/builder-api -s at-once --bind router:builder-router.default
DEBUG 2018-06-07T22:55:14Z: habitat_common::ui: UI { shell: Shell { input: InputStream { isatty: true }, out: OutputStream { coloring: Auto, isatty: true, is_colored(): true, supports_color(): true }, err: OutputStream { coloring: Auto, isatty: true, is_colored(): true, supports_color(): true } } }
DEBUG 2018-06-07T22:55:14Z: hab: clap cli args: ["hab", "svc", "load", "habitat/builder-api", "-s", "at-once", "--bind", "router:builder-router.default"]
DEBUG 2018-06-07T22:55:14Z: hab: remaining cli args: []
DEBUG 2018-06-07T22:55:14Z: hab::config: No CLI config found, loading defaults
DEBUG 2018-06-07T22:55:14Z: tokio::reactor: scheduling direction for: 0
DEBUG 2018-06-07T22:55:14Z: tokio::reactor: loop process - 1 events, 0.000s
DEBUG 2018-06-07T22:55:14Z: tokio::reactor: scheduling direction for: 0
DEBUG 2018-06-07T22:55:14Z: tokio::reactor: loop process - 1 events, 0.000s
DEBUG 2018-06-07T22:55:14Z: tokio::reactor: dropping I/O source: 0
DEBUG 2018-06-07T22:55:14Z: tokio::reactor: loop process - 1 events, 0.000s
✗✗✗
✗✗✗ Connection closed
✗✗✗
++ echo -n 1
[31][default:/src:1]#
```

There's nothing in the supervisor log. Also the router bind seems properly exported:
```
[42][default:/src:0]# curl localhost:9631/services/builder-router/default/config
{"listen":"0.0.0.0","log_level":"info","port":5562}
```

Additional weirdness: when `start-builder router` runs, it correctly installs and then runs `hab svc start`:
```
✗✗✗
✗✗✗ [Err: 2] habitat/builder-router not loaded.
✗✗✗
» Installing habitat/builder-router
☁ Determining latest version of habitat/builder-router in the 'stable' channel
☛ Verifying habitat/builder-router/7114/20180302152133
…
✓ Installed habitat/builder-router/7114/20180302152133
★ Install of habitat/builder-router/7114/20180302152133 complete with 2 new packages installed.
The habitat/builder-router service was successfully loaded
```

But then `start-builder api` doesn't even do the install:
```
[28][default:/src:0]# start-builder api
+ start-builder api
+ stop-on-failure _start-builder api
+ set -e
+ _start-builder api
+ [[ 1 -eq 0 ]]
+ for component in '"$@"'
+ '[' -v 'svc_params[api]' ']'
+ load-if-not-loaded habitat/builder-api -s at-once --bind router:builder-router.default
+ stop-on-failure _load-if-not-loaded habitat/builder-api -s at-once --bind router:builder-router.default
+ set -e
+ _load-if-not-loaded habitat/builder-api -s at-once --bind router:builder-router.default
+ local pkg_ident
+ pkg_ident=habitat/builder-api
+ hab sup status habitat/builder-api
+ grep --color=auto habitat/builder-api
✗✗✗
✗✗✗ [Err: 2] habitat/builder-api not loaded.
✗✗✗
+ hab svc load habitat/builder-api -s at-once --bind router:builder-router.default
✗✗✗
✗✗✗ Connection closed
✗✗✗
+ rv=1
+ '[' 1 -ne 0 ']'
+ echo -e '\033[0;31mERROR: _start-builder api aborted due to error\033[0m'
ERROR: _start-builder api aborted due to error
```

A manual install succeeds:
```
[29][default:/src:1]# hab pkg install habitat/builder-api
+ hab pkg install habitat/builder-api
» Installing habitat/builder-api
☁ Determining latest version of habitat/builder-api in the 'stable' channel
…
✓ Installed habitat/builder-api/7370/20180526002244
★ Install of habitat/builder-api/7370/20180526002244 complete with 1 new packages installed.
```

But same behavior when I try to run it afterward:
```
[34][default:/src:0]# hab svc load habitat/builder-api -s at-once --bind router:builder-router.default
✗✗✗
✗✗✗ Connection closed
✗✗✗
```

Here's how I'm reproducing the behavior:
```
➤ ~/habitat/target/debug/hab --version
hab 0.57.0-dev
```
```
➤ hab studio rm
```
```
➤ env HAB_STUDIO_SUP=false CI_OVERRIDE_CHANNEL=unstable ~/habitat/target/debug/hab studio enter
```
```
[1][default:/src:0]# hab pkg install core/hab-sup/0.57.0-dev/20180607214134
```
^ for [the `hab svc status <IDENT>` fix](https://github.com/habitat-sh/habitat/pull/5166)
```
[3][default:/src:1]# sup-run 
[4][default:/src:0]# start-builder
``` 

I'm thinking it must be a bug in the supervisor or `hab` binary.

**Comment by jonlives** [comment]
This seems to have been fixed in 0.57.0 anecdotally - I could reproducibly hit this on 0.56.1 though.

**Comment by baumanj** [comment]
Thanks, @jonlives! That's strange, but good to know.
